packages/job-dispatch/src/target.ts

```
@@ -109,73 +250,24 @@ export const upsertTargets = async (
-      .from(targetMetadata)
-      .where(
-        inArray(
-          targetMetadata.targetId,
-          targets.map((t) => t.id),
-        ),
+      .then((targets) =>
+        targets.map((t) => ({
+          ...t,
+          ...targetsToInsert.find(
+            (t) =>
+              t.identifier === t.identifier && t.workspaceId === t.workspaceId,
```

Biome

[error] Comparing to itself is potentially pointless.



(lint/suspicious/noSelfCompare)